app/src/main/java/com/example/myfirstapp/ShoppingTabFragment.java

```diff
@@ -65,7 +65,10 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle bu
         editShoppingItemPrice.setHint(R.string.edit_text_add_price);
         editShoppingItemPrice.setInputType(InputType.TYPE_CLASS_NUMBER | InputType.TYPE_NUMBER_FLAG_DECIMAL);
 
+        // Add a listener to add a $ when a user starts to type
         editShoppingItemPrice.addTextChangedListener(new TextWatcher() {
+            boolean ignoreLoop = false;
```

Comment:
Flags like this tend to be a code smell. Usually because there is a more idiomatic way of achieving what you want to achieve.

You are def on the right track trying to prevent an infinite loop here, I just wonder if there is a different way where you don't have to rely on a flag.